### PR TITLE
Cross-build 0.23 for Scala.js (rebased)

### PR DESCRIPTION
**Pull request by armanbilge**
Did some rebasing of #5285 to try and better preserve history. Opening a separate PR instead of force-pushing in case of screwups.

### CODE REVIEW

File: core/shared/src/main/scala/org/http4s/util/UrlCoding.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*
```
Comment (armanbilge) [inline core/shared/src/main/scala/org/http4s/util/UrlCoding.scala:1]: Darn, somehow git still couldn't track this file.

File: client/shared/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala [hunk 2]
```diff
@@ -19,10 +19,7 @@ package client
```
Comment (armanbilge) [inline client/shared/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala:19]: Did work for `ClientRouteTestBattery` 👍

**Comment by armanbilge** [comment]
@rossabaker ready when you are!

File: core/shared/src/main/scala/org/http4s/util/UrlCoding.scala (hunk 1 above)
Comment (rossabaker) [reply core/shared/src/main/scala/org/http4s/util/UrlCoding.scala:1]: That's fine.

**Pull request merged**